bigcode_eval/evaluator.py

```diff
@@ -38,11 +40,15 @@ def __init__(self, accelerator, model, tokenizer, args):
         # code evaluation permission
         self.allow_code_execution = args.allow_code_execution
 
-    def generate_text(self, task_name):
+    def generate_text(self, task_name, intermediate_generations=None):
         task = tasks.get_task(task_name, self.args)
         dataset = task.get_dataset()
         # if args.limit is None, use all samples
         n_tasks = self.args.limit if self.args.limit else len(dataset)
+        # when args.limit is None
+        # adjust n_tasks by args.limit_start to prevent out of bounds issues 
+        if not self.args.limit:
+            n_tasks -= self.args.limit_start
```

Comment:
IIUC the issue is that setting `limit_start` to a positive value and `limit` to `None` causes an out-of-bound issue?
Shouldn't we instead add a check after parsing the args that this is not the case?

Reply:
Oh, I guess for your test you wanted to set a `limit_start`, but without a `limit`?

Reply:
Yep that's correct, the case I'm envisioning is say you want to start from the 50th sample until the end.

In the current method, we'd have to manually calculate `len(dataset) - args.limit_start` in order to calculate and set `args.limit` manually to align the indices. This way would be much easier if we set `args.limit = None` and automatically calculate how many samples to eval on 😄